src/main/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingService.kt

```diff
@@ -71,8 +83,66 @@ class PrintResponseProcessingService(
         }
     }
 
+    private fun processBatchResponsesMySql(batchResponses: List<BatchResponse>) {
+        batchResponses.forEach { batchResponse ->
+            val certificates =
+                certificateRepository.findByStatusAndPrintRequestsBatchId(
+                    SENT_TO_PRINT_PROVIDER,
+                    batchResponse.batchId
+                )
+            processBatchCertificates(batchResponse, certificates)
+            certificateRepository.saveAll(certificates)
+        }
+    }
+
+    private fun processBatchCertificates(batchResponse: BatchResponse, certificates: List<Certificate>) {
+        with(batchResponse) {
+            val newStatus =
+                if (status == SUCCESS) RECEIVED_BY_PRINT_PROVIDER else PENDING_ASSIGNMENT_TO_BATCH
+
+            certificates.forEach {
+                it.addStatus(
+                    status = newStatus,
+                    eventDateTime = timestamp.toInstant(),
+                    message = message
+                )
+
+                if (status != SUCCESS) {
+                    val currentPrintRequest = it.getCurrentPrintRequest()
+                    currentPrintRequest.requestId = idFactory.requestId()
+                    currentPrintRequest.batchId = null
+                }
+            }
+        }
+    }
+
     fun processPrintResponse(printResponse: ProcessPrintResponseMessage) {
```

Comment:
Does this method need to be `@Transactional`?

Reply:
Yes. definitely - thanks for spotting 👍